Sources/DataPipeline/plugins/DeviceAttributes.swift

```diff
@@ -6,24 +6,31 @@ class DeviceAttributes: Plugin {
     public let type = PluginType.before
     public weak var analytics: Analytics?
 
+    public var token: String?
     public var attributes: [String: Any]?
 
     public required init() {}
 
     public func execute<T: RawEvent>(event: T?) -> T? {
-        guard var workingEvent = event else { return event }
+        guard var workingEvent = event,
+              var context = workingEvent.context?.dictionaryValue
+        else { return event }
```

Comment:
What is this doing? From reading it, I don't understand what this code is for?

Reply:
transforming the data, if you look into the `payload` expected by CDP and generated by SDK that might help in understand it more. 

`Raw Event` already has a json, which has `context` and other `json object` in it. This plugin updates the values of `context` json object.